packages/dashboard/src/util/maybe.ts

```diff
@@ -0,0 +1,35 @@
+enum MaybeType {
+  Just = 'maybe-type__just',
+  Nothing = 'maybe-type__nothing',
+}
+
+export interface Just<T> {
+  type: typeof MaybeType.Just;
+  value: T;
+}
+
+export interface Nothing {
+  type: typeof MaybeType.Nothing;
+}
+
+export type Maybe<T> = Just<T> | Nothing;
+
+export const Nothing = (): Nothing => ({
+  type: MaybeType.Nothing,
+});
+
+export const Just = <T>(value: T): Just<T> => ({
+  type: MaybeType.Just,
+  value,
+});
+
+export const isJust = <T>(m: Maybe<T>): m is Just<T> => m.type === MaybeType.Just;
+
+export const maybeWithDefault = <T>(defaultVal: T, m: Maybe<T>): T => {
+  switch (m.type) {
+    case MaybeType.Nothing:
+      return defaultVal;
+    case MaybeType.Just:
+      return m.value;
+  }
+};
```

Comment:
If you make a public util, treat it like an API and document its intended usage.